### PR TITLE
Introduce HxController

### PR DESCRIPTION
**Pull request by ia3andy**

### CODE REVIEW

**Comment by ia3andy** [comment]
I am adding tests

File: runtime/src/main/java/io/quarkiverse/renarde/htmx/HxController.java [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+
+import io.quarkiverse.renarde.Controller;
+import io.quarkus.qute.Qute;
+import io.quarkus.qute.TemplateInstance;
+import io.vertx.core.http.HttpServerResponse;
+
+/**
+ * This is a Controller's overlay to simplify usage of Quarkus Renarde alongside {@see <a href="https://htmx.org/">htmx</a>}.
+ */
+public class HxController extends Controller {
```
Comment (FroMage) [inline runtime/src/main/java/io/quarkiverse/renarde/htmx/HxController.java:19]: Shouldn't this be `abstract`?

File: deployment/src/main/java/io/quarkiverse/renarde/deployment/RenardeProcessor.java [hunk 2]
```diff
@@ -720,6 +723,11 @@ private Map<DotName, AnnotationInstance>[] getParameterAnnotations(MethodInfo me
         return parameterAnnotations;
     }
 
+    @BuildStep
+    void removeHxController(BuildProducer<ExcludedControllerBuildItem> excludedControllerBuildItems) {
+        excludedControllerBuildItems.produce(new ExcludedControllerBuildItem(DOTNAME_HX_CONTROLLER));
```
Comment (FroMage) [inline deployment/src/main/java/io/quarkiverse/renarde/deployment/RenardeProcessor.java:728]: Why did you need to remove it? Is it because it's not `abstract`?

**Review by FroMage — COMMENTED** [review]
Looks great, do you mind adding some docs to the manual?

File: runtime/src/main/java/io/quarkiverse/renarde/htmx/HxController.java (hunk 1 above)
Comment (ia3andy) [reply runtime/src/main/java/io/quarkiverse/renarde/htmx/HxController.java:19]: @FroMage If I remember well I reproduced Controller which is not abstract right? But yes I could switch to abstract..

Comment (FroMage) [reply runtime/src/main/java/io/quarkiverse/renarde/htmx/HxController.java:19]: Well, OK, I see. But, `Controller` never needed to be abstract because it's not a controller itself :) It's not even annotated with `@Path` so, there's nothing special.

Your `HxController` is a subtype of `Controller` so it gets picked up to be a controller and a `@Path` will be added to it by our processor.

Which is why you're excluding it.

Now that I looked at the processor, we still scan abstract controllers in case they define public endpoints. Which is not the case for `HxController`. So, probably I'd say it's better to be safe and make both `Controller` and `HxController` `abstract` and keep this one as `excluded` as you did, since you're not defining endpoints, and are also not using any redirects.

File: docs/modules/ROOT/pages/index.adoc [hunk 3]
```diff
@@ -1098,6 +1098,38 @@ You can read values from the Flash scope in your Qute views using the `{flash:na
 
 The Flash scope only lasts from one request to the next and is cleared at each request.
 
+== htmx
+
+If you want to use htmx with Renarde, we added some an extra layer to make things more convenient:
```
Comment (FroMage) [inline docs/modules/ROOT/pages/index.adoc:1103]:
```suggestion
If you want to use htmx with Renarde, we added an extra layer to make things more convenient:
```

Also, can you add a pointer to HTMX docs?

**Review by FroMage — CHANGES_REQUESTED** [review]
Looks much better :)

**Review by FroMage — APPROVED** [review]

**Comment by FroMage** [comment]
Perfect!
